Softeq.XToolkit.Common.Droid.Tests/Extensions/EditTextExtensionsTests/EditTextExtensionsTests.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Developed by Softeq Development Corporation
+// http://www.softeq.com
+
+using Android.Text;
+using Android.Widget;
+using Softeq.XToolkit.Common.Droid.Extensions;
+using Softeq.XToolkit.Common.Droid.TextFilters;
+using Xunit;
+
+namespace Softeq.XToolkit.Common.Droid.Tests.Extensions.EditTextExtensionsTests
+{
+    public class EditTextExtensionsTests
+    {
+        [Fact]
+        public void SetFilters_WhenCalledWithoutFilters_DoNothing()
+        {
+            EditText editText = new EditText(MainActivity.Current);
+            editText.SetFilters();
+
+            var appliedFilters = editText.GetFilters();
+            if (appliedFilters != null)
+            {
+                Assert.Empty(appliedFilters);
+            }
+        }
+
+        [Fact]
+        public void SetFilters_WhenCalledWithMultipleFilters_AppliesAllFilters()
+        {
+            EditText editText = new EditText(MainActivity.Current);
+
+            var filter1 = new ForbiddenCharsInputFilter(new char[] { 'a' });
+            var filter2 = new ForbiddenCharsInputFilter(new char[] { 'b' });
+            var filter3 = new ForbiddenCharsInputFilter(new char[] { 'a' });
+            var filters = new IInputFilter[] { filter1, filter2, filter3 };
+
+            editText.SetFilters(filter1, filter2, filter3);
+
+            var appliedFilters = editText.GetFilters();
+            Assert.Equal(filters, appliedFilters);
+        }
+
+        [Fact]
+        public void SetFilters_WhenCalledWithMultipleFiltersWithNullAndDuplicates_AppliesAllFilters()
+        {
+            EditText editText = new EditText(MainActivity.Current);
+
+            var filter1 = new ForbiddenCharsInputFilter(new char[] { 'a' });
```

Comment:
Prefer to avoid dependencies on another custom filter. Seems better to use a mock object for this.

Reply:
Fixed